python/lsst/cp/pipe/cpFunctions.py

```diff
@@ -0,0 +1,516 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If
+
+# Utility tasks/methods copied from pipe_drivers/constructCalibs.py
+
+import datetime
+import numpy as np
+import time
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+import lsst.afw.math as afwMath
+import lsst.afw.image as afwImage
+import lsst.daf.base as dafBase
+
+
+class BlessCalibration(pipeBase.Task):
+    """Create a way to bless existing calibration products.
+
+    The inputs are assumed to have been constructed via cp_pipe, and
+    already exist in the butler.
+
+    Parameters
+    ----------
+    kwargs : `dict`
+        Parameters to initialize the blessing process.  Keys:
+
+        ``"butler"``
+            Data butler to use.
+        ``"inputCollection"``
+            Data collection to pull calibrations from.
+        ``"outputCollection"``
+            Data collection to store final calibrations.
+    """
+    def __init__(self, **kwargs):
+        self.butler = kwargs['butler']
+        self.registry = self.butler.registry
+
+        self.inputCollection = kwargs['inputCollection']
+        self.outputCollection = kwargs['outputCollection']
+
+        self.calibrationLabel = ''
+        self.instrument = ''
+
+    def findInputs(self, datasetTypeName, inputTypeString=None):
+        """Find and prepare inputs for blessing.
+
+        Parameters
+        ----------
+        datasetTypeName : `str`
+            Dataset that will be blessed.
+        inputTypeString : `str`
+            Dataset name for the input datasets.
+
+        Raises
+        ------
+        RuntimeError :
+            Raised if no input datasets found or if the calibration
+            label exists and is not empty.
+        """
+        if inputTypeString is None:
+            inputDatasetTypeName = datasetTypeName + "Proposal"
+
+        self.inputValues = list(self.registry.queryDatasets(inputDatasetTypeName,
+                                                            collections=[self.inputCollection]))
+        if len(self.inputValues) == 0:
+            raise RuntimeError(f"No inputs found for dataset {inputDatasetTypeName} "
+                               f"in {self.inputCollection}")
+
+        # Construct calibration label and choose instrument to use.
+        self.calibrationLabel = "{}/{}".format(datasetTypeName,
+                                               self.inputCollection)
```

Comment:
Why split the line?
And you clearly know and like f-strings, so why not make this `f"{datasetTyopeName}/{self.inputCollection}"`?